tests/kernel/mem_protect/stackprot/src/main.c

```diff
@@ -85,15 +101,23 @@ void alternate_thread(void *p1, void *p2, void *p3)
 	ARG_UNUSED(p3);
 
 	TC_PRINT("Starts %s\n", __func__);
-	check_input(__func__,
-		    "Input string is too long and stack overflowed!\n");
+
+	if (should_fail) {
+		check_input_fail(__func__,
+				"Input string is too long 2222 and stack overflowed!\n");
+	} else {
+		check_input(__func__,
+				"Input string is too long and stack overflowed!\n");
+	}
 	/*
 	 * Expect this thread to terminate due to stack check fail and will not
 	 * execute pass here.
 	 */
 	print_loop(__func__);
 
-	ret = TC_FAIL;
+	if (!should_fail) {
```

Comment:
```suggestion
	if (should_fail == 0) {
```
(Forgive me for being mobile and can't see more context.)

Reply:
I am curious about this suggestion. `should_fail` is a boolean type, so AFAIK `!should_fail` is a valid construction according with our and MISRA-C code guidelines. In other hand, comparing boolean with integer is probably violating MISRA-C.

Reply:
Oh, you can ignore my suggestion. I mistakenly thought it was int :)